rust/src/data_structures.rs

```diff
@@ -52,7 +52,7 @@ impl AppData {
 /// # Notes on usage
 /// If you use "0.0.0.0" or "::" as ip value, then you need to also provide
 /// `announced_address`.
-#[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Deserialize, Serialize)]
```

Comment:
I had to remove `Ord` and `PartialOrd` because `RangeInclusive` doesn't implement them. No downsides AFAIS.